src/pkg/lint/findings.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package lint contains functions for verifying zarf yaml files are valid
+package lint
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/fatih/color"
+)
+
+// PackageFinding is a struct that contains a finding about something wrong with a package
+type PackageFinding struct {
+	// YqPath is the path to the key where the error originated from, this is sometimes empty in the case of a general error
+	YqPath      string
+	Description string
+	// Item is the value of a key that is causing an error, for example a bad image name
+	Item string
+	// PackageNameOverride shows the name of the package that the error originated from
+	// If it is not set the base package will be used when displaying the error
+	PackageNameOverride string
+	// PackagePathOverride shows the path to the package that the error originated from
+	// If it is not set the base package will be used when displaying the error
+	PackagePathOverride string
+	Severity            Severity
+}
+
+// Severity is the type of finding
+type Severity int
+
+// different severities of package errors
+const (
+	SevErr Severity = iota + 1
+	SevWarn
+)
+
+func (f PackageFinding) itemizedDescription() string {
+	if f.Item == "" {
+		return f.Description
+	}
+	return fmt.Sprintf("%s - %s", f.Description, f.Item)
+}
+
+func colorWrapSev(s Severity) string {
+	if s == SevErr {
+		return message.ColorWrap("Error", color.FgRed)
+	} else if s == SevWarn {
+		return message.ColorWrap("Warning", color.FgYellow)
+	}
+	return "unknown"
+}
+
+func filterLowerSeverity(findings []PackageFinding, severity Severity) []PackageFinding {
+	findings = helpers.RemoveMatches(findings, func(finding PackageFinding) bool {
+		return finding.Severity > severity
+	})
+	return findings
+}
+
+// PrintFindings prints the findings of the given severity in a table
+func PrintFindings(findings []PackageFinding, severity Severity, baseDir string, packageName string) {
```

Comment:
I know that it is tricky but this would be nice to defer to the cli calling. If it is very difficult to do right now you can ignore this.

Reply:
Yeah it's difficult to do right now as we'd have to change create & dev to return findings to the cli so it could then be printed. I think I'm leaning towards doing that. Would love to chat synchronously on the implications of implementing that.